src/octoprint/plugins/backup/__init__.py

```diff
@@ -129,6 +123,18 @@ def get_template_configs(self):
             {"type": "wizard", "name": gettext("Restore Backup?")},
         ]
 
+    ##~~ Utility Methods
+
+    def build_backup_filename(self, backup_prefix=None):
```

Comment:
As this is being used from both CLI and full context, this should better be a class method that takes a `settings` object as parameter and operates on that. 

`self._settings` will not be defined in case of the CLI context, so it should not be access here. Turning the method from instance into class method with explicit `settings` parameter makes that clearer.

Reply:
I think I have updated this to what you mean, sorry for my lack of proper python programming.